File: internal/pkg/openstack_helm/monitoring.go
package openstack_helm

type Monitoring struct {
	Prometheus `yaml:"prometheus,omitempty"`
}

type Prometheus struct {
	Enabled bool `yaml:"enabled,omitempty"`
}

var MemcachedMonitoring Monitoring = Monitoring{
	Prometheus: Prometheus{
		Enabled: true,
	},
}

File: atmosphere/models/openstack_helm/monitoring.py
from schematics import types

from atmosphere.models import base


class PrometheusMonitoring(base.Model):
    enabled = types.BooleanType()


class Monitoring(base.Model):
    prometheus = types.ModelType(PrometheusMonitoring)

    @classmethod
    def for_chart(cls, chart):
        if chart == "memcached":
            return Monitoring(
                {
                    "prometheus": PrometheusMonitoring(
                        {
                            "enabled": True,
                        }
                    )
                }
            )
